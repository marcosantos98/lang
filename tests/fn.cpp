typedef const char* cstr;
extern int puts(cstr);
int sum(int, int);
void ___entry___();
int sum(int a, int b){
}
void ___entry___(){
}
int main(int argc, char** argv) {___entry___();}