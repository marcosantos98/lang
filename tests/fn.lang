extern "c" puts :: int(str cstr)

sum :: int(a int, b int) {
    return a + b
}

main :: void() {}
