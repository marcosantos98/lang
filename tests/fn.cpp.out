/* -- Builtin types -- */
typedef const char* cstr;
/* -- ------------- -- */
/* --------------- Builtin ---------------- */
struct String {
    void* data;
    int len;
};
static String ___empty_string___ = {(void*)"", 0};

String make_string(const char* cstr, int len) {
    return {(void*)cstr, len};
}

int len_str(String s) {
    return s.len;
}
/* --------------- Builtin ---------------- */

extern "C" int puts(cstr);
int sum(int, int);
void ___entry___();

int sum(int a, int b){
return a+b;
}
void ___entry___(){
}
int main(int argc, char** argv) {
___entry___();
}

